### PR TITLE
chore: add ability to publish to GitHub registry

### PR DESCRIPTION
**Pull request by booc0mtaco**
### Summary:

- add in config to publish to github registry upon release
- will test after config is validated

### Test Plan:

- [ ] Wrote [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
- [ ] CI tests / new tests are not applicable
- [ ] Manually tested my changes, but I want to keep the details secret
- [x] Manually tested my changes, and here are the details:
  - test run on PR to see publish results

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+          # Configure NPM to point to Github's package registry.
+          registry-url: https://npm.pkg.github.com/
```
Comment (ahuth) [inline .github/workflows/release.yml:26]:
```suggestion
          node-version-file: '.node-version'
          cache: 'yarn',
          # Configure NPM to point to Github's package registry.
          registry-url: https://npm.pkg.github.com/
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Install deps
+        uses: bahmutov/npm-install@v1
```
[the remark this answers is not in the document]
Comment (booc0mtaco) [reply .github/workflows/release.yml:29]: @ahuth I'll restore that pull_request option and test these out

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Install Dependencies 🧶
+        run: yarn install --immutable
+
+      - name: Build EDS 🛠️
+        run: yarn build
+
+      # Publish the package on Github's package registry for internal CZI use. You should still
+      # publish to the npm registry separately from this job.
+      - name: Publish package to the Github registry
+        run: npm publish --tag alpha --foreground-scripts true
```
Comment (timzchang) [inline .github/workflows/release.yml:39]: checking my understanding: the `--tag alpha` means every version of eds we publish to the GH registry will be an alpha version, is that right? 

Could you explain why if this is intended? thanks!

**Comment by booc0mtaco** [comment]
got the publish to work by simply removing `postpack`. Noted that in the husky docs, when using `npm`, you only specify `prepare` but the setup for `yarn` is more complex. Also, the output from the script does seem to indicate `postpack` runs before publish, so it appears that it, in fact, WAS running before publish.

Will try `postpublish` instead of `postpack`.

**Comment by booc0mtaco** [comment]
Trying this configuration: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages

**Review by andremalan — APPROVED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -1,6 +1,11 @@
 name: Build & Test CI
 
-on: push
+on:
+  push:
+  workflow_call:
+    secrets:
+      CHROMATIC_PROJECT_TOKEN:
+        required: true
```
Comment (jeremiah-clothier) [inline .github/workflows/test.yml:8]: What does adding the `required` here do? Just error out sooner in ci?

Comment (booc0mtaco) [reply .github/workflows/test.yml:8]: i think so. before we were reusing this config to run all tests for releases. Andrew and I found that was redundant given how our lifecycle is set up now. 

We could remove this, but if we ever need to reuse this workflow in another workflow, we are set up to fail if the keys aren't passed over

**Pull request merged**